compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp

```diff
@@ -330,7 +330,19 @@ SmallVector<BufferOp> createBuffers(OpBuilder &builder,
   SmallVector<BufferOp> buffers;
   if (deviceModel.isShimTile(tile.getCol(), tile.getRow())) return buffers;
   auto fifoType = cast<AIEObjectFifoType>(createOp.getElemType());
-  auto elemType = cast<MemRefType>(fifoType.getElementType());
+  MemRefType fifoElemType = cast<MemRefType>(fifoType.getElementType());
+  ArrayRef<int64_t> shape = fifoElemType.getShape();
+  Type elType = fifoElemType.getElementType();
+
+  Attribute memSpace = [&]() -> Attribute {
```

Comment:
If I look at the tests, I don't see any memory space attributes ? I don't see how that happens, because here you do assign memory space attributes?

Reply:
The tests are not lowered through iree-amd-aie, our tiling strategy introduces these memory spaces. I haven't added a test for this, I guess I should. 

Here I'm only assigning a memory space attribute sometimes. It's just correcting the memory space if there is already one present.

Reply:
I see. To avoid the hardcoding below based on row number, I think we should instead just remove all memory space attributes. That was the intention of the logic before (although that logic standardized on memSpace == 1).

Reply:
That would definitely be nicer. A bit more work, but I'll give it a try.